src/lib/core/TLVWriter.cpp

```diff
@@ -694,19 +658,13 @@ CHIP_ERROR TLVWriter::WriteElementHead(TLVElementType elemType, Tag tag, uint64_
     ABORT_ON_UNINITIALIZED_IF_ENABLED();
 
     VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INCORRECT_STATE);
-    uint8_t * p;
-    uint8_t stagingBuf[17]; // 17 = 1 control byte + 8 tag bytes + 8 length/value bytes
-
     if (IsContainerOpen())
         return CHIP_ERROR_TLV_CONTAINER_OPEN;
 
+    uint8_t stagingBuf[17]; // 17 = 1 control byte + 8 tag bytes + 8 length/value bytes
+    uint8_t * p     = stagingBuf;
     uint32_t tagNum = TagNumFromTag(tag);
 
-    if ((mRemainingLen >= sizeof(stagingBuf)) && (mMaxLen >= sizeof(stagingBuf)))
-        p = mWritePoint;
```

Comment:
So this was trying to avoid the extra copy if we have enough space to just write directly, right?  Removing this does simplify the logic, so as long as it's not an efficiency problem...  This code is called a lot when writing TLV.

Reply:
- The amount of TLV written is comparatively small, compared to TLV read.
- The special handling of the path is just more code that can be done wrong. Blitting the bytes is very fast, compared to all the rest of the logic.